pkg/handler/collector/oci/oci.go

```diff
@@ -0,0 +1,229 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oci
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/logging"
+	"github.com/pkg/errors"
+	"github.com/regclient/regclient"
+	"github.com/regclient/regclient/types/manifest"
+	"github.com/regclient/regclient/types/ref"
+)
+
+const (
+	OCICollector = "OCICollector"
+)
+
+type ociCollector struct {
+	repo          string
+	tag           string
+	checkedDigest []string
+	poll          bool
+	interval      time.Duration
+}
+
+func NewOCICollector(ctx context.Context, repo string, tag string, poll bool, interval time.Duration) *ociCollector {
```

Comment:
For registries that support list images in the registry, should we allow for downloading of all metadata associated with all images in that registry? Or do you imagine that would be a different collector that calls out to this one similar to the GitCollector calling the FileCollector

Reply:
Sure we can add support for lists of images in a registry. We can have 3 options.
1 - specify the registry and it will list and pull from all repositories in that registry (keeping track of each tag in each repo via a `map[string][]string`  (polling or one-time call for all repos and tags in the registry)
2 - specify the repo and it will list all the tags and poll on that. Getting new tags as they appear (polling only)
3 - specify the repo and the tag and it will collect the metadata associated with just that tag (one-time run only)

Reply:
Ons second thought. It might be cleaner to do it with another registry collector calling this one for each repo.

Reply:
@mlieberman85 spoke with @sudo-bmitch further regarding this and he suggested that this would be more painful and bug probe as `_catalog` is not implemented the same across all registries. Each cloud-hosted registry adds its own custom API for listing things within a single org. We can revisit this later if needed but oci collector is currently setup such that it can be called by an upstream "registry collector" to collect from all repos in a registry in the future if needed.

Reply:
Cool, can you just add a comment in the OCI collector just calling that out? I just want folks to have something devs can look at in the case of someone trying to understand why is isn't there.

Reply:
the only possible concern i have here is needing to have many thread context running if used this way.. it may end up with the different collectors not having control over the queries and may hit quota limits and livelock each other. I am wondering if making it a `repoTags map[string][]string` of repos to tags helps with this?

Reply:
I see what you mean. I will change it to take in `repoTags map[string][]string` so multiple repos and tags can be specified.

Reply:
done.